gateway.go

```diff
@@ -69,7 +69,12 @@ func NewGateway(opts ...GatewayOption) *StarknetGateway {
 func (sg *StarknetGateway) newRequest(
 	ctx context.Context, method, endpoint string, body interface{},
 ) (*http.Request, error) {
-	req, err := http.NewRequestWithContext(ctx, method, sg.Feeder+endpoint, nil)
+	url := sg.Feeder+endpoint
+	if strings.Contains(endpoint, "add_transaction") {
```

Comment:
`HasSuffix` would work here i think? minor improvement but safer

```suggestion
	if strings.HasSuffix(endpoint, "add_transaction") {
```